common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -166,7 +166,7 @@ func parseSystemSearchAttributeValues(name string, value any) (any, error) {
 		}
 	case searchattribute.ExecutionStatus:
 		if status, isNumber := value.(int64); isNumber {
-			value = enumspb.WorkflowExecutionStatus_name[int32(status)]
+			value = enumspb.WorkflowExecutionStatus(status).String()
```

Comment:
I was curious how it handles the case for unknown statuses and saw this:
```
default:
    // Deprecated: Use WorkflowExecutionStatus.Descriptor instead.
    return strconv.Itoa(int(x))
```
It's probably fine, but I wanted to call that out.